File: tests/human/x/fs/break_continue.fs
open System

exception Break
exception Continue

let numbers = [1; 2; 3; 4; 5; 6; 7; 8; 9]

try
    for n in numbers do
        try
            if n % 2 = 0 then raise Continue
            if n > 7 then raise Break
            printfn "odd number: %d" n
        with Continue -> ()
with Break -> ()

File: tests/machine/x/pl/break_continue.pl
:- style_check(-singleton).
:- initialization(main, main).
main :-
    Numbers = [1, 2, 3, 4, 5, 6, 7, 8, 9],
    catch(
        (
            member(N, Numbers),
                catch(
                    (
                        (((N mod 2) == 0) ->
                            throw(continue),
                            true
                        ; true
                        ),
                        ((N @> 7) ->
                            throw(break),
                            true
                        ; true
                        ),
                        write("odd number:"),
                        write(' '),
                        write(N),
                        nl,
                        true
                    ), continue, true),
                    fail
                ; true
            ), break, true),
            true.
